server/src/test/java/io/crate/integrationtests/disruption/discovery/DiskDisruptionIT.java

```diff
@@ -0,0 +1,228 @@
+            while (stopGlobalCheckpointFetcher.get() == false) {
+                try {
+                    var response = sqlExecutor.execute("select id, seq_no_stats['global_checkpoint'] " +
+                                                       "from sys.shards where table_name='test'", null).actionGet();
+                    for (var row : response.rows()) {
+                        final int shardId = (int) row[0];
+                        final long globalCheckpoint = (long) row[1];
+                        shardToGcp.compute(shardId, (i, v) -> Math.max(v, globalCheckpoint));
+                    }
+                } catch (Exception e) {
+                    // ignore
+                    logger.debug("failed to fetch shard stats", e);
+                }
+            }
+        });
+
+        globalCheckpointSampler.start();
+
+        try (BackgroundIndexer indexer = new BackgroundIndexer("test", "data", sqlExecutor, -1, RandomizedTest.scaledRandomIntBetween(2, 5),
+                                                               false, random())) {
+            indexer.setRequestTimeout(TimeValue.ZERO);
+            indexer.setIgnoreIndexingFailures(true);
+            indexer.setFailureAssertion(e -> {});
+            indexer.start(-1);
+
+            waitForDocs(randomIntBetween(1, 100), indexer, "test");
+
+            logger.info("injecting failures");
+            injectTranslogFailures();
+            logger.info("stopping indexing");
+        }
+
+        logger.info("full cluster restart");
+        internalCluster().fullRestart(new InternalTestCluster.RestartCallback() {
+
+            @Override
+            public void onAllNodesStopped() {
+                logger.info("stopping failures");
+                stopTranslogFailures();
+            }
+
+        });
+
+        stopGlobalCheckpointFetcher.set(true);
+
+        logger.info("waiting for global checkpoint sampler");
+        globalCheckpointSampler.join();
+
+        logger.info("waiting for green");
+        ensureGreen();
+
+       execute("select distinct id, seq_no_stats['max_seq_no'] from sys.shards where table_name='test' and " +
+               "routing_state in ('STARTED', 'RELOCATING')");
+
+        assertThat(response.rowCount(), is((long) numberOfShards));
+
+        for (var row : response.rows()) {
+            final int shardId = (int) row[0];
+            final long maxSeqNo = (long) row[1];
+            assertThat(maxSeqNo, greaterThanOrEqualTo(shardToGcp.get(shardId)));
+        }
+    }
+
+    /**
+     * Waits until at least a give number of document is visible for searchers
+     *
+     * @param numDocs number of documents to wait for
+     * @param indexer a {@link org.elasticsearch.test.BackgroundIndexer}. It will be first checked for documents indexed.
+     *                This saves on unneeded searches.
+     */
+    public void waitForDocs(final long numDocs, final BackgroundIndexer indexer, String table) throws Exception {
```

Comment:
Any reason why this method is not added to the `ESIntegTestCase` like in the ES upstream?